src/container/stack.ts

```diff
@@ -0,0 +1,19 @@
+import { interfaces } from "../interfaces/interfaces";
+
+export class Stack<T> implements interfaces.Stack<T> {
+  private _stack: T[] = [];
+  push(entry: T): void {
+      this._stack.push(entry);
+  }
+  pop(): T | undefined {
+      return this._stack.pop();
+  }
+
+  peek(): T | undefined {
```

Comment:
Can we maybe change this function name ? 
I don't think `peek` implies getting last element in an array

Reply:
Array is the implementation detail.  Peek is what you would have got from pop without removing it ?